### ISSUE TITLE
$cfg['TablePrimaryKeyOrder'] = 'desc';

### ISSUE THREAD

**Issue by OlafvdSpek** [phpmyadmin/phpmyadmin#16832]
```
$cfg['TablePrimaryKeyOrder'] = 'desc';
```

Could this be made the default?

**Comment by williamdes** [comment]
I would say no 🤔 
```php
/**
 * Default sort order by primary key.
 *
 * @global string $cfg['TablePrimaryKeyOrder']
 */
$cfg['TablePrimaryKeyOrder'] = 'NONE';
```

cc @mauriciofauth for advice

**Comment by OlafvdSpek** [comment]
Why not? ;)

**Comment by nunoperalta** [comment]
@williamdes - what would be really useful to me would be that when I click on a column name, the default is to sort DESC.

99% of the times I want to sort a column, it is DESC, but I always have to press twice, because the first sort is ASC.
This is inconvenient many times, especially when the query is expensive (full table scan).

I was looking at the tooltip that explains the shortcuts, but there doesn't seem to be anything (like pressing Shift/etc) that makes the sort DESC.

Any ideas? Or there's no implementation for that at the moment? (I can create a feature request)

**Comment by nunoperalta** [comment]
Well, there's the "Order" config that can be set to "DESC".

The SMART option only does it for TIME, DATE, DATETIME and TIMESTAMP.
However, I use Unix timestamps (UNSIGNED INT) for Time, so this option was never useful to me in the past.

I changed it to "DESC" to try, but the columns seem to be still sorting ascending by default. Not sure if it's a bug or not...

**Comment by williamdes** [comment]
This setting exists :) https://docs.phpmyadmin.net/en/latest/config.html#cfg_TablePrimaryKeyOrder

I am not in favor of changing any default value of config variables, but I am open to feature requests ;)

**Comment by nunoperalta** [comment]
That's only for Primary Key, isn't it? I'm saying when I click on some other column to sort. I'd like if the default could be DESC.

![image](https://user-images.githubusercontent.com/487079/115994223-f9a2b200-a5cd-11eb-8af9-1872177f90bb.png)

**Comment by williamdes** [comment]
Yes, seems to be only for the primary key. That makes sense with the naming of the config variable.
Adding one more config variable seems like an idea
https://github.com/phpmyadmin/phpmyadmin/blob/51e17b81233a267e0d8c3fc9d2e0051f6ca666dd/libraries/classes/Sql.php#L539-L559

**Comment by nunoperalta** [comment]
Thank you. But, isn't what this one is for?

https://docs.phpmyadmin.net/en/latest/config.html#cfg_Order

Just doesn't seem to be working (what I said above).

**Comment by OlafvdSpek** [comment]
> I changed it to "DESC" to try, but the columns seem to be still sorting ascending by default. Not sure if it's a bug or not...

That'd be a bug, wouldn't it? Could you file a bug report for it?

**Comment by nunoperalta** [comment]
#16851 16851

**Comment by williamdes** [comment]
It works when you click on a table name to browse it, but seems only to work in some cases.
Here is one:
```sql
START TRANSACTION;
SET time_zone = "+00:00";

CREATE TABLE `source` (
  `id` varchar(64) COLLATE utf8mb4_unicode_ci NOT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci COMMENT='the source' ROW_FORMAT=COMPACT;


INSERT INTO `source` (`id`) VALUES
('01593cc4acf2177c59a6aa934ff7cb3511ea78bf9fb073ba547ecc4941996106');

ALTER TABLE `source`
  ADD PRIMARY KEY (`id`);
COMMIT;
```

**Comment by williamdes** [comment]
Initial feature: 7969336d04177cd58cc1e4fc7210ee99bedb9d49

Source: https://sourceforge.net/p/phpmyadmin/feature-requests/1502/

**Comment by OlafvdSpek** [comment]
> Yes, seems to be only for the primary key. That makes sense with the naming of the config variable.
> Adding one more config variable seems like an idea
> https://github.com/phpmyadmin/phpmyadmin/blob/51e17b81233a267e0d8c3fc9d2e0051f6ca666dd/libraries/classes/Sql.php#L539-L559

This seems broken when the primary key spans multiple columns.. the code would only sort on the last column in the key.

**Comment by williamdes** [comment]
About what you said @OlafvdSpek I think I found the reason while researching for #18130
See : https://github.com/phpmyadmin/phpmyadmin/issues/16851#issuecomment-1437447487

What do you think @OlafvdSpek?
Does the feature request for a smarter SMART mode + bug fixing the Order setting solve this issue?
